File: SecureChatBackend/Domain/Entities/Conversation.cs
using System;

namespace SecureChatBackend.Domain.Entities;

public sealed class Conversation
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public bool IsGroup { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? LastMessageAt { get; set; }

    public ICollection<ConversationParticipant> Participants { get; set; } = new List<ConversationParticipant>();
    public ICollection<Message> Messages { get; set; } = new List<Message>();
}

File: SecureChatMobileApp/types/conversation.ts
export type ConversationParticipant = {
  userId: string;
  publicKey?: string | null;
  sessionId?: string | null;
  username?: string | null;
};

export type ConversationRecord = {
  id: string;
  isGroup?: boolean;
  createdAt: string;
  lastMessageAt?: string | null;
  participants: ConversationParticipant[];
};
